### PR TITLE
Replace StringBuffer by String

### PR DESCRIPTION
**Pull request by DanVanAtta**
IDE fix: Reports any usages of java.lang.StringBuffer and java.lang.StringBuilder which can be replaced with a single java.lang.String concatenation. Using a String concatenation makes the code shorter and simpler. This inspection only reports when the resulting concatenation is at least as efficient or more efficient than the original StringBuffer or StringBuilder code.

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/framework/EngineVersionProperties.java [hunk 1]
```diff
@@ -98,29 +98,28 @@ String getChangeLogLink() {
...
+        + "\">" + getLinkToDownloadLatestVersion() + "</a>"
+        + "<br />Backup Mirror: <a class=\"external\" href=\"" + getLinkAltToDownloadLatestVersion()
+        + "\">" + getLinkAltToDownloadLatestVersion() + "</a>"
+        + "<br /><br />Please note that installing a new version of TripleA will not remove any old copies of "
+        + "TripleA."
+        + "<br />So be sure to either manually uninstall all older versions of TripleA, or change your "
+        + "shortcuts to the new TripleA."
+        + "<br /><br />What is new:<br />"
+        + "</html>";
+    return text;
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/EngineVersionProperties.java]: This method should just return the string directly, no need to create a variable first.

File: game-core/src/main/java/games/strategy/engine/framework/EngineVersionProperties.java [hunk 2]
```diff
@@ -98,29 +98,28 @@ String getChangeLogLink() {
...
-    final StringBuilder text = new StringBuilder("<html><body>");
-    text.append("Link to full Change Log:<br /><a class=\"external\" href=\"").append(getChangeLogLink()).append("\">")
-        .append(getChangeLogLink()).append("</a><br />");
-    text.append("</body></html>");
-    return text.toString();
+    final String text =
+        "<html><body>" + "Link to full Change Log:<br /><a class=\"external\" href=\"" + getChangeLogLink() + "\">"
+            + getChangeLogLink() + "</a><br />"
+            + "</body></html>";
+    return text;
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/EngineVersionProperties.java]: Same here

File: game-core/src/main/java/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java [hunk 3]
```diff
@@ -515,18 +515,17 @@ String getStatus() {
...
+        + "\n\nStatus:\n"
+        + getStatus()
+        + "\n\nServer:\n"
+        + getServerModel()
+        + "\n\n"
+        + DebugUtils.getThreadDumps()
+        + "\n\n"
+        + DebugUtils.getMemory()
+        + "\n\nDump finished.\n";
+    System.out.println(sb);
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java]: No need for a variable here as well + the variable name isn't fitting either

File: game-core/src/main/java/games/strategy/engine/framework/startup/mc/ClientModel.java [hunk 4]
```diff
@@ -471,16 +471,15 @@ public Action getHostBotGetGameSaveClientAction(final Component parent) {
...
-    return sb.toString();
+    final String sb = "ClientModel GameData:" + (gameDataOnStartup == null ? "null" : gameDataOnStartup.getGameName())
+        + "\n"
+        + "Connected:" + (messenger == null ? "null" : messenger.isConnected()) + "\n"
+        + messenger
+        + "\n"
+        + remoteMessenger
+        + "\n"
+        + channelMessenger;
+    return sb;
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/startup/mc/ClientModel.java]: Var name isn't fitting, no need to create a var

File: game-core/src/main/java/games/strategy/triplea/util/UnitCategory.java [hunk 5]
```diff
@@ -108,11 +108,10 @@ public int hashCode() {
...
   public String toString() {
-    final StringBuilder sb = new StringBuilder();
-    sb.append("Entry type:").append(type.getName()).append(" owner:").append(owner.getName()).append(" damaged:")
-        .append(damaged).append(" bombingUnitDamage:").append(bombingDamage).append(" disabled:").append(disabled)
-        .append(" dependents:").append(dependents).append(" movement:").append(movement);
-    return sb.toString();
+    final String sb = "Entry type:" + type.getName() + " owner:" + owner.getName() + " damaged:"
+        + damaged + " bombingUnitDamage:" + bombingDamage + " disabled:" + disabled
+        + " dependents:" + dependents + " movement:" + movement;
+    return sb;
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/triplea/util/UnitCategory.java]: Name isn't fitting, no need to assign a variable first

**Comment by DanVanAtta** [comment]
It is a much nicer cleanup with the intermediate variables removed @RoiEXLab ; that is done: 9562d01

Please have a look when you have a chance.

**Pull request merged**